docs/source/examples/beam.py

```diff
@@ -47,6 +47,8 @@ def __init__(self, N=4, **kwargs):
         theta_eq[0] = (th[0] >= th_base)  # base boundary condition
         displ_eq = (w >= w.left + 0.5*dx*(th + th.left))
         displ_eq[0] = (w[0] >= w_base)
+        for constraint in [shear_eq, moment_eq, theta_eq, displ_eq]:
+            constraint.reset_varkeys()
```

Comment:
better maybe, but I still have the backwards compatibility hesitation.

What about having array constraint's `__setitem__` call reset_varkeys? Doesn't that get directly at the issue here?